### PR TITLE
♻️(frontend) refactor CourseProductRelation types

### PR DESCRIPTION
**Pull request by rlecellier**
## Purpose

To improve performance, we had to change what's CourseProductRelation
endpoint return.
Therefor we need to refactor this endpoint usage in our frontend.

### CODE REVIEW

File: src/frontend/js/hooks/useCourseProductUnion/index.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export const useCourseProductUnion = ({
     queryAConfig: {
       queryKey: ['user', 'courses'],
       fn: api.courses.get,
-      filters: { organization_id: organizationId },
+      filters: { organization_id: organizationId, has_listed_course_runs: true },
```
Comment (rlecellier) [inline src/frontend/js/hooks/useCourseProductUnion/index.ts:45]: @FIXME: when adding this filter, joanie return duplicated courses. Twice each course.

File: src/frontend/js/settings.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export const REACT_QUERY_SETTINGS = {
     throttleTime: 500,
   },
   // Cache is garbage collected after this delay
-  cacheTime: 24 * 60 * 60 * 1000, // 24h in ms
+  cacheTime: 0, // 24 * 60 * 60 * 1000, // 24h in ms
```
Comment (jbpenrath) [inline src/frontend/js/settings.ts]: Do not forgot to rollback this change

File: src/frontend/js/types/Joanie.ts [hunk 3]
```diff
@@ -101,16 +89,25 @@ export interface Product {
...
   order: Nullable<OrderLite>;
   target_courses: TargetCourse[];
 }
 
+export interface CourseProductRelationCourse {
+  id: string;
+  code: string;
+  title: string;
+  cover?: JoanieFile;
+}
```
Comment (jbpenrath) [inline src/frontend/js/types/Joanie.ts:105]: In order to improve maintainability, what do you think to herit this type from `AbstractCourse`

```suggestion
export type CourseProductRelationCourse = Pick<AbstractCourse, 'id' | 'code' | 'title' | 'cover'>
```

File: src/frontend/js/components/CourseGlimpse/utils.ts [hunk 4]
```diff
@@ -27,11 +27,26 @@ const getCourseGlimpsePropsFromCourseProductRelation = (
...
+        }
+      : null,
+    // TODO (rlecellier): We should use owner_organization instead of these organization that are selling ones.
+    organization: {
+      title: courseProductRelation.organizations[0].title,
+      image: courseProductRelation.organizations[0].logo
+        ? {
+            src: courseProductRelation.organizations[0].logo.src,
+          }
+        : null,
```
Comment (jbpenrath) [inline src/frontend/js/components/CourseGlimpse/utils.ts:45]: I think you should pass the whole logo object. In order to pass `src` and `srcset` properties to `img` tag

**Review by PanchoutNathan — APPROVED** [review]

**Pull request merged**